### ISSUE TITLE
Wrong formatting for windows_service

### ISSUE THREAD

**Issue by morro-tr** [DataDog/ansible-datadog#402]
Using the below yaml part for windows_service produces wrong conf.yml after deploying through Ansible.

```
datadog_checks: 
  windows_service:
    instances:
      - services:
          - xxxxx
```

Produce the following windows_service.d\conf.yml :

```
instances:
-   services:
    - xxxxx
```

This leads to the configuration for the windows service integration failing.
reproduces every time on Datadog agent 7, using Ansible Galaxy 2.9.4.

**Comment by bkabrda** [comment]
Hi 👋 thanks for the report! I just tested this, replacing `xxxxx` by `Dnscache` and it works correctly for me (it's true that the resulting yaml format looks a bit weird with the extra spaces before `services`, but it is a correct yaml).

Could you please let me know how exactly the integration is failing? E.g. logs/tracebacks from the agent log that show what exactly is going wrong. Thanks!

**Comment by morro-tr** [comment]
Hi!
I've checked further and it seems that after restarting the agent the issue is solved, so the problem as you mentioned is not the YAML structure.
I have a ticket with Datadog, I don't know if you have access to the ticketing system, the number is - 586634 and I sent a new Flare from a server that I ran, only after restarting the agent the windows_service started working.

**Comment by bkabrda** [comment]
Ok, so do I understand it correctly that the problem is that when you add the check configuration via ansible, the agent doesn't pick it up after the ansible run, but you have to restart it manually once more for the check configuration to get picked up?

In the case that config file for any check is changed, the task `Create a configuration file for each Datadog check` should notify `restart datadog-agent-win` so in theory this should be enough. If you can reproduce this, how exactly does it happen? Do you add/modify the configuration of the check to your playbook, then rerun ansible and you don't see any data from this check? If that's the case, do you see the task `Create a configuration file for each Datadog check` as `changed` by ansible?

Feel free to let me know here or in the customer case if you're more comfortable using that, I'm ok with using any of these (and I do have access there).

**Comment by morro-tr** [comment]
Hi,

This is for a new Datadog agent installation.
It's reproducing for every new host I try to run it on.

It does show -
10:28:30  TASK [datadog.datadog : Create a configuration file for each
Datadog check] ****
10:28:30  Sunday 07 November 2021  08:28:30 +0000 (0:00:01.609)
0:03:53.518 *******
10:28:35  changed: [xxxxxxxx] => (item=windows_service)

But at the end -
10:28:54  RUNNING HANDLER [datadog.datadog : restart datadog-agent-win]
******************
10:28:54  Sunday 07 November 2021  08:28:54 +0000 (0:00:00.021)
0:04:17.270 *******
10:28:54  skipping: [xxxxxxxx]

I've added a manual task to restart the datadog service which works, but I
believe this issue should be fixed through Datadog Ansible.

**Comment by bkabrda** [comment]
Ok, thanks for the additional info. I'll look into it.

**Comment by bkabrda** [comment]
Hmm, I can't reproduce this. Couple more questions:

* What version of ansible-datadog are you using?
* What version of ansible are you using?
* Is it possible that your playbook contains `datadog_enabled: false`?
* Are you by chance running ansible in check mode?
* If possible, could you provide your playbook that reproduces the behavior? (Obviously after stripping out any secrets that it might contain.)

Thanks!

**Comment by morro-tr** [comment]
I believe it might have been the `datadog_enabled: false`.
Right now we simply a task for an additional restart to the agent no matter what.
Thanks for the assistance.

**Comment by bkabrda** [comment]
Ok, thanks for letting me know. Feel free to open a new issue if you experience more problems!
